include/solvers/time_harmonic_maxwell.h

```diff
@@ -0,0 +1,728 @@
+  write_checkpoint() override;
+
+  /**
+   * @brief Set solution vector of Auxiliary Physics using checkpoint
+   */
+  virtual void
+  read_checkpoint() override;
+
+  /**
+   * @brief Returns a vector of references to TableHandler objects that needs to
+   * be serialized/deserialized for a given TimeHarmonicMaxwell solver.
+   *
+   * @return Structure containing the TableHandler objects and their corresponding file names.
+   */
+  virtual std::vector<OutputStructTableHandler>
+  gather_tables() override;
+
+  /**
+   * @brief Compute the Kelly error estimator used to refine mesh on a auxiliary physic parameter.
+   *
+   * @param ivar The current element of the map simulation_parameters.mesh_adaptation.variables
+   *
+   * @param estimated_error_per_cell The deal.II vector of estimated_error_per_cell
+   */
+  virtual void
+  compute_kelly(const std::pair<const Variable,
+                                Parameters::MultipleAdaptationParameters> &ivar,
+                dealii::Vector<float> &estimated_error_per_cell) override;
+
+  /**
+   * @brief Compute the DPG error estimator based on the energy norm residual used to refine mesh of the TimeHarmonicMaxwell physic.
+   *
+   * @param ivar The current element of the map simulation_parameters.mesh_adaptation.variables
+   *
+   * @param estimated_error_per_cell The deal.II vector of estimated_error_per_cell
+   */
+  virtual void
+  compute_energy_norm(
+    const std::pair<const Variable, Parameters::MultipleAdaptationParameters>
+                          &ivar,
+    dealii::Vector<float> &estimated_error_per_cell);
+
+  /**
+   * @brief Sets-up the DofHandler and the degree of freedom associated with the physics.
+   */
+  virtual void
+  setup_dofs() override;
+
+  /**
+   * @brief Sets-up the initial conditions associated with the physics. Generally, physics
+   * only support imposing nodal values, but some physics additionally support
+   * the use of L2 projection or steady-state solutions.
+   */
+  virtual void
+  set_initial_conditions() override;
+
+  /**
+   * @brief Set up preconditioner. Not used for the auxiliary physics but
+   * needed for the compilation of the non-linear solver.
+   */
+  void
+  setup_preconditioner() override;
+
+  /**
+   * @brief Define the constraints for the TimeHarmonicMaxwell solver. This is equivalent to the define_non_zero_constraints method in other physics solvers, but here there is no need to separate zero and non-zero constraints.
+   */
+  void
+  define_constraints();
+
+  /**
+   * @brief Call for the solution of the linear system of equation using CG solver.
+   */
+  void
+  solve_linear_system() override;
+
+  /**
+   * @brief Getter method to access the private attribute dof_handler for the
+   * physic currently solved. NB : The dof_handler that is returned is the one
+   * for the interior trial space only has it is where the solution lives. The
+   * other two DoFHandlers of the class are used for computation only.
```

Comment:
I would explain earlier in the file that there is a solution on the skeleton and one inside each cell for the trial functions, and one test function. I would also rephrase the second sentence for it to reflect what you mean. 
```suggestion
   * @brief Getter method to access the private attribute dof_handler for the
   * physic currently solved. NB : The dof_handler that is returned is the one
   * for the interior trial space only, where the solution lives. The
   * other two DoFHandlers of the class are used for computation only.
```